typing/ctype.ml

```diff
@@ -592,35 +592,34 @@ let rec filter_row_fields erase = function
 type variable_kind = Row_variable | Type_variable
 exception Non_closed of type_expr * variable_kind
 
-(* [free_vars] collects the variables of the input type expression. It
+(* [free_vars] walks over the variables of the input type expression. It
    is used for several different things in the type-checker, with the
    following bells and whistles:
    - If [env] is Some typing environment, types in the environment
      are expanded to check whether the apparently-free variable would vanish
      during expansion.
-   - We collect both type variables and row variables, paired with
-     a [variable_kind] to distinguish them.
    - We do not count "virtual" free variables -- free variables stored in
      the abbreviation of an object type that has been expanded (we store
      the abbreviations for use when displaying the type).
 
-   [free_vars] returns a [(variable * bool) list], while
-   [free_variables] below drops the type/row information
-   and only returns a [variable list].
+   [free_vars] accumulates its answer in a monoid-like structure, with
+   an initial element [zero] and a combining function [add_one], passing
+   [add_one] information about whether the variable is a normal type variable
+   or a row variable.
  *)
-let free_vars ?env mark ty =
+let free_vars ~zero ~add_one ?env mark ty =
```

Comment:
This is nitpicking, but if you look at `ListLabels`, the label for accumulators is `init:`.

Reply:
> `zero:` sounds a bit algebraic

Yes! If I were writing Haskell, I would have parameterized this function over a Monoid. (Sadly there is no good support in either language for a _commutative_ monoid.) This (to me) makes it clear that the function just accumulates a bunch of answers and gloms them together. So “algebraic” was exactly the look I was going for.

That said, I think consistency is good, and I’ll change this to `init` later today.